src/librbd/api/Group.cc

```diff
@@ -1223,13 +1223,60 @@ int Group<I>::snap_list(librados::IoCtx& group_ioctx, const char *group_name,
   for (auto snap : cls_snaps) {
     snaps->push_back(
 	group_snap_info_t {
-	   snap.name,
-	   static_cast<group_snap_state_t>(snap.state)});
+          snap.id,
+          snap.name,
+          static_cast<group_snap_state_t>(snap.state)});
 
   }
   return 0;
 }
 
+template <typename I>
+int Group<I>::snap_info(librados::IoCtx& group_ioctx, const char *group_name,
+                        const char *snap_name,
+                        group_snap_info_t *group_snap,
+                        std::vector<group_image_snap_info_t> *image_snaps)
+{
+  std::vector<cls::rbd::GroupSnapshot> cls_group_snaps;
+
+  int r = group_snap_list(group_ioctx, group_name, &cls_group_snaps);
+  if (r < 0) {
+    return r;
+  }
+
+  cls::rbd::GroupSnapshot *cls_group_snap = nullptr;
+  for (auto &i : cls_group_snaps) {
+    if (i.name == string(snap_name)) {
```

Comment:
No need to create a temporary `std::string`:
```suggestion
    if (i.name == snap_name)) {
```